pkg/webhooks/podimagespec/podimagespec.go

```diff
@@ -0,0 +1,293 @@
+	if err != nil {
+		log.Error(err, "Unable mutate pod")
+		return admissionctl.Errored(http.StatusInternalServerError, err)
+	}
+
+	return admissionctl.PatchResponseFromRaw(request.Object.Raw, mutatedPod)
+}
+
+// renderPod renders the Pod in the admission Request
+func (s *PodImageSpecWebhook) renderPod(request admissionctl.Request) (*corev1.Pod, error) {
+	decoder, err := admissionctl.NewDecoder(s.s)
+	if err != nil {
+		return nil, err
+	}
+	pod := &corev1.Pod{}
+	err = decoder.Decode(request, pod)
+	if err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
+
+func podContainsContainerRegexMatch(pod *corev1.Pod) (podMatch bool) {
+	podMatch = false
+
+	for i := range pod.Spec.Containers {
+		containerMatch, namespace, _, _ := checkContainerImageSpecByRegex(pod.Spec.Containers[i].Image)
+		if containerMatch && namespace == "openshift" {
+			podMatch = true
+		}
+	}
+
+	for i := range pod.Spec.InitContainers {
+		containerMatch, namespace, _, _ := checkContainerImageSpecByRegex(pod.Spec.InitContainers[i].Image)
+		if containerMatch && namespace == "openshift" {
+			podMatch = true
+		}
+	}
+
+	return
+}
+
+func (s *PodImageSpecWebhook) mutatePod(pod *corev1.Pod, ctx context.Context) ([]byte, error) {
+	mutatedPod := pod.DeepCopy()
+
+	for i := range pod.Spec.Containers {
+		imageURI, err := s.lookupImageStreamTagSpec(pod.Spec.Containers[i].Image, ctx)
+		if err != nil {
+			return []byte{}, err
+		}
+		mutatedPod.Spec.Containers[i].Image = imageURI
+	}
+
+	for i := range pod.Spec.InitContainers {
+		imageURI, err := s.lookupImageStreamTagSpec(pod.Spec.InitContainers[i].Image, ctx)
+		if err != nil {
+			return []byte{}, err
+		}
+		mutatedPod.Spec.InitContainers[i].Image = imageURI
+	}
+
+	return json.Marshal(mutatedPod)
+}
+
+// checkImageRegistryStatus checks the status of the image registry service
+func (s *PodImageSpecWebhook) checkImageRegistryStatus(ctx context.Context) (bool, error) {
+	var err error
+	registryV1 := &registryv1.Config{}
+
+	err = s.kubeClient.Get(ctx, client.ObjectKey{Name: "cluster"}, registryV1)
+	if err != nil {
+		return false, fmt.Errorf("failed to get image registry config: %v", err)
+	}
+
+	// if image registry is set to managed then it is operational
+	if registryV1.Spec.ManagementState == operatorv1.Managed {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// checkContainerImageSpecByRegex checks to see if the image is in the openshift namespace in the internal registry
+func checkContainerImageSpecByRegex(imagespec string) (bool, string, string, string) {
+	matches := imageRegex.FindStringSubmatch(imagespec)
+	if matches == nil {
+		return false, "", "", ""
+	}
+	namespaceIndex := imageRegex.SubexpIndex("namespace")
+	imageIndex := imageRegex.SubexpIndex("image")
+	tagIndex := imageRegex.SubexpIndex("tag")
+	return true, matches[namespaceIndex], matches[imageIndex], matches[tagIndex]
+}
+
+func (s *PodImageSpecWebhook) lookupImageStreamTagSpec(imagespec string, ctx context.Context) (string, error) {
```

Comment:
```suggestion
func (s *PodImageSpecWebhook) lookupImageStreamTagSpec(ctx context.Context, imagespec string) (string, error) {
```

nit, by convention context should be the first argument